### PR TITLE
feat: generate contribute page needed static data

### LINKED ISSUE #759: The /contribute page broken if current IP hits GitHub API's limitation
GitHub API has frequency now, while some regions are not able to access the host properly, can we move it to Build CI for that? Will we make this change in the next PR?

_Originally posted by @bzp2010 in https://github.com/apache/apisix-website/pull/758#discussion_r753925848_

### PR DESCRIPTION
**Pull request by SkyeYoung**
Fixes: #759

Changes:

- add generate-repos-info.js and do related changes

Screenshots of the change:

none

### CODE REVIEW

**Comment by SkyeYoung** [comment]
@bzp2010 @juzhiyuan It seems that the construction of netlify is independent. But now the contribute page needs to rely on `repos-info.json`, netlify is not configured to run the command to generate this file 😅. 

What should I do?

**Comment by juzhiyuan** [comment]
> @bzp2010 @juzhiyuan It seems that the construction of netlify is independent. But now the contribute page needs to rely on `repos-info.json`, netlify is not configured to run the command to generate this file 😅.
> 
> What should I do?

Hi, usually when we meet this case, we will run it locally to test 😅 But if you like, I could invite you to Netlify's Team account, you could adjust the Netlify configuration.

**Comment by SkyeYoung** [comment]
> Hi, usually when we meet this case, we will run it locally to test

Emm, in fact, I have solved this problem by adding a default JSON file that will be replaced after construction. And, please review the changes😸

**Review by juzhiyuan — DISMISSED** [review]
LGTM from logics, please format the `generate-repos-info.js` file with 2 spaces.

**Comment by SkyeYoung** [comment]
> please format the `generate-repos-info.js` file with 2 spaces.

I'll do it now.

**Review by juzhiyuan — DISMISSED** [review]
LGTM

File: website/repos-info.json [hunk 1]
```diff
@@ -0,0 +1 @@
..."description":"the docker for Apache APISIX","Star":338,"Watch":35,"Fork":227},"issues":[{"comments":0,"number":200,"title":"feat request: build dashboard docker image on travis ci ARM machine","htmlUrl":"https://github.com/apache/apisix-docker/issues/200"}]},"apache/apisix-java-plugin-runner":{"info":{"description":"APISIX Plugin Runner in Java","Star":57,"Watch":25,"Fork":40},"issues":[{"comments":1,"number":92,"title":"request help: How to get all the information in the original request header.","htmlUrl":"https://github.com/apache/apisix-java-plugin-runner/issues/92"},{"comments":5,"number":75,"title":"feature: java plugin log path error","htmlUrl":"https://github.com/apache/apisix-java-plugin-runner/issues/75"},{"comments":3,"number":49,"title":"[discuss]: increase the timeout of socket communication between apisix and java plugin.","htmlUrl":"https://github.com/apache/apisix-java-plugin-runner/issues/49"}]},"apache/apisix-go-plugin-runner":{"info":{"description":"Go Plugin Runner for APISIX","Star":66,"Watch":27,"Fork":29},"issues":[]},"apache/apisix-python-plugin-runner":{"info":{"description":"Apache APISIX Python plugin runner","Star":15,"Watch":20,"Fork":13},"issues":[]}}
```
Comment (bzp2010) [inline website/repos-info.json]: In `info` section, we should keep the key in lower case. Like, `Star`, `Watch`, `Fork`.

File: scripts/generate-repos-info.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+const repoList = require("../website/config/docs").map(v => v.githubRepo)
+const res = {}
+
+const tasks = new listr([
+  {
+    title: "Fetch repos' info and good first issues",
+    task: ()=> new listr(repoList.map((repo)=>{
+      return {
+        title: `Fetch ${repo}`,
+        task: ()=>new listr([
```
Comment (bzp2010) [inline scripts/generate-repos-info.js]: Formatting code required.

**Review by Baoyuantop — DISMISSED** [review]
nice work

File: website/repos-info.json (hunk 1 above)
Comment (bzp2010) [inline website/repos-info.json]: And, maybe we can change each data value to 0 default.

File: scripts/generate-repos-info.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+          {
+            title: `${repo}'s info`,
+            task: () => axios.get(`https://api.github.com/repos/${repo}`, axiosConfig)
```
Comment (bzp2010) [inline scripts/generate-repos-info.js]: We should consider the case of a failed data pull, at which point we might be to interrupt the CI execution.

File: scripts/generate-repos-info.js (hunk 2 above)
Comment (SkyeYoung) [reply scripts/generate-repos-info.js]: Sorry, do we have any lint?

File: website/repos-info.json (hunk 1 above)
Comment (SkyeYoung) [reply website/repos-info.json]: **Each** prop? I will rewrite this part manually.😅

File: scripts/generate-repos-info.js (hunk 3 above)
Comment (SkyeYoung) [reply scripts/generate-repos-info.js]: I use this part.
```js
tasks.run()
  .then(()=> {
    console.log("[Finish] Generate repos' info and good first issues finished")
  })
  .catch((err)=>{
    console.error(err)
    process.exit(1)
  })
```
Is it not enough? I'm not familiar with this part. I'll check other files.

Comment (bzp2010) [reply scripts/generate-repos-info.js]: Yes, I think so. When exiting with a non-zero status code should cause the CI to report an error and interrupt.

File: website/repos-info.json (hunk 1 above)
Comment (bzp2010) [reply website/repos-info.json]: I mean to generate an empty data. Both info values and issue lists can be written as empty lists.

File: scripts/generate-repos-info.js (hunk 2 above)
Comment (bzp2010) [reply scripts/generate-repos-info.js]: We don't have a lint script for javascript scripts yet, so you can rely on the IDE's code formatting features for now.

Comment (SkyeYoung) [reply scripts/generate-repos-info.js]: > We don't have a lint script for javascript scripts yet, so you can rely on the IDE's code formatting features for now.

Ok

**Review by juzhiyuan — DISMISSED** [review]
Would better put repos-info.json into a static or config or temp folder 🤔

**Comment by SkyeYoung** [comment]
> Would better put repos-info.json into a static or config or temp folder 🤔

It seems that there is no suitable folder. we don't have a temp folder. Maybe `config` folder is a not bad choice? I'll do it now.

**Review by juzhiyuan — APPROVED** [review]

**Review by bzp2010 — APPROVED** [review]

**Pull request merged**